### ISSUE TITLE
Operator broken after cluster restart

### ISSUE THREAD

**Issue by liyimeng** [minio/operator#651]
I deploy minio operator 4.0.4 some weeks ago. Today I have a power failure at home, which cause my computers restarted. 

After that, the operator seem broken, it flooding logs like these in k3s serivce

```
I0525 00:55:56.640204   10936 request.go:655] Throttling request took 1.044048862s, request: GET:https://127.0.0.1:6444/apis/minio.min.io/v2?timeout=32s
E0525 00:55:57.227864   10936 cacher.go:419] cacher (*unstructured.Unstructured): unexpected ListAndWatch error: failed to list minio.min.io/v1, Kind=Tenant: conversion webhook for minio.min.io/v2, Kind=Tenant failed: Post "https://operator.minio.svc:4222/webhook/v1/crd-conversion?timeout=30s": x509: certificate signed by unknown authority; reinitializing...

```

And operator logs is also flooding logs like this: 

```
2021/05/24 23:00:26 http: TLS handshake error from 10.42.1.0:15618: remote error: tls: bad certificate
2021/05/24 23:00:26 http: TLS handshake error from 10.42.0.0:43631: remote error: tls: bad certificate
2021/05/24 23:00:26 http: TLS handshake error from 10.42.2.1:36853: remote error: tls: bad certificate
2021/05/24 23:00:27 http: TLS handshake error from 10.42.1.0:42813: remote error: tls: bad certificate
2021/05/24 23:00:27 http: TLS handshake error from 10.42.0.0:11237: remote error: tls: bad certificate
2021/05/24 23:00:27 http: TLS handshake error from 10.42.2.1:22147: remote error: tls: bad certificate
2021/05/24 23:00:28 http: TLS handshake error from 10.42.1.0:12598: remote error: tls: bad certificate
2021/05/24 23:00:28 http: TLS handshake error from 10.42.0.0:58986: remote error: tls: bad certificate
2021/05/24 23:00:28 http: TLS handshake error from 10.42.2.1:55676: remote error: tls: bad certificate
2021/05/24 23:00:29 http: TLS handshake error from 10.42.1.0:52745: remote error: tls: bad certificate
2021/05/24 23:00:29 http: TLS handshake error from 10.42.0.0:35806: remote error: tls: bad certificate
2021/05/24 23:00:29 http: TLS handshake error from 10.42.2.1:34713: remote error: tls: bad certificate
2021/05/24 23:00:30 http: TLS handshake error from 10.42.1.0:19389: remote error: tls: bad certificate
2021/05/24 23:00:30 http: TLS handshake error from 10.42.0.0:8505: remote error: tls: bad certificate
2021/05/24 23:00:30 http: TLS handshake error from 10.42.2.1:45395: remote error: tls: bad certificate
2021/05/24 23:00:31 http: TLS handshake error from 10.42.1.0:6343: remote error: tls: bad certificate
2021/05/24 23:00:31 http: TLS handshake error from 10.42.0.0:23413: remote error: tls: bad certificate
2021/05/24 23:00:31 http: TLS handshake error from 10.42.2.1:31151: remote error: tls: bad certificate
2021/05/24 23:00:32 http: TLS handshake error from 10.42.1.0:30223: remote error: tls: bad certificate
2021/05/24 23:00:32 http: TLS handshake error from 10.42.0.0:8906: remote error: tls: bad certificate
2021/05/24 23:00:32 http: TLS handshake error from 10.42.2.1:6589: remote error: tls: bad certificate
2021/05/24 23:00:33 http: TLS handshake error from 10.42.1.0:5722: remote error: tls: bad certificate
2021/05/24 23:00:33 http: TLS handshake error from 10.42.0.0:41675: remote error: tls: bad certificate
2021/05/24 23:00:33 http: TLS handshake error from 10.42.2.1:39266: remote error: tls: bad certificate
2021/05/24 23:00:34 http: TLS handshake error from 10.42.1.0:7618: remote error: tls: bad certificate
2021/05/24 23:00:34 http: TLS handshake error from 10.42.0.0:15011: remote error: tls: bad certificate
2021/05/24 23:00:34 http: TLS handshake error from 10.42.2.1:36986: remote error: tls: bad certificate
2021/05/24 23:00:35 http: TLS handshake error from 10.42.1.0:15995: remote error: tls: bad certificate
2021/05/24 23:00:35 http: TLS handshake error from 10.42.0.0:1900: remote error: tls: bad certificate
2021/05/24 23:00:35 http: TLS handshake error from 10.42.2.1:8554: remote error: tls: bad certificate
2021/05/24 23:00:36 http: TLS handshake error from 10.42.1.0:16878: remote error: tls: bad certificate
2021/05/24 23:00:36 http: TLS handshake error from 10.42.0.0:37823: remote error: tls: bad certificate
2021/05/24 23:00:36 http: TLS handshake error from 10.42.2.1:3715: remote error: tls: bad certificate
2021/05/24 23:00:37 http: TLS handshake error from 10.42.1.0:42227: remote error: tls: bad certificate
2021/05/24 23:00:37 http: TLS handshake error from 10.42.0.0:17321: remote error: tls: bad certificate
2021/05/24 23:00:37 http: TLS handshake error from 10.42.2.1:1053: remote error: tls: bad certificate
2021/05/24 23:00:38 http: TLS handshake error from 10.42.1.0:30596: remote error: tls: bad certificate
2021/05/24 23:00:38 http: TLS handshake error from 10.42.0.0:59344: remote error: tls: bad certificate
2021/05/24 23:00:38 http: TLS handshake error from 10.42.2.1:27719: remote error: tls: bad certificate

```

My disks are going to filled with these crazy log very soon, i guess.  How do I stop it?

## Expected Behavior

I expect operator survive and recover after a k8s cluster restart. 

## Current Behavior

Operator stop working, seem related to some certification issue. 

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Version used (`minio-operator`): 4.0.4
* Environment name and version (e.g. kubernetes v1.17.2): v1.20
* Server type and version:  desktop 
* Operating System and version (`uname -a`): ubuntu
* Link to your deployment file:

**Comment by harshavardhana** [comment]
Mostly your k8s cluster is broken @liyimeng - did you configure proper backups for your etcd? in your k8s setup?

**Comment by dvaldivia** [comment]
@liyimeng this looks like the TLS certificate used by operator is no longer valid and MinIo's are trying to talk to Operator with a bad TLS certificate, an option is deleting the `operator-tls` secret from the operator namespace, and restarting oeprator so it re-generates the TLS certificate, then copying that secret to all namespaces that have a tenant

**Comment by liyimeng** [comment]
> Mostly your k8s cluster is broken @liyimeng - did you configure proper backups for your etcd? in your k8s setup?

My cluster seem working for other deployments. I have a lot of other deployments. They work as it were after power is back.  So I am not sure if minio operator is more sensitive or other reasons.

**Comment by liyimeng** [comment]
@dvaldivia I delete operator deployment and operator-tls. Then redeploy operator, operator-tls is recreated as you suggested.  However, still get the same error. 

The connection seems come from a web hook somewhere 

````
E0525 12:00:44.037863   10936 cacher.go:419] cacher (*unstructured.Unstructured): unexpected ListAndWatch error: 

failed to list minio.min.io/v1, Kind=Tenant: conversion webhook for minio.min.io/v2, Kind=Tenant failed: Post "https://operator.minio.svc:4222/webhook/v1/crd-conversion?timeout=30s": 

x509: certificate signed by unknown authority; reinitializing...
```


Right now I only have a tenants in minio namespace, which is actually sharing with operator. 

I know k3s rotate its root cert if the service get restarted and old cert is about to expired.  Is operator dealing with cert rotation in some way?

**Comment by harshavardhana** [comment]
I think your cluster is hosed after power failure, I think there will be subtle things that will be broken. 

Unfortunately your cluster looks like was not protected enough has lost internal states.
```
E0525 12:00:44.037863   10936 cacher.go:419] cacher (*unstructured.Unstructured): unexpected ListAndWatch error: 
```

**Comment by harshavardhana** [comment]
Closing this issue since there is nothing that the operator can do to fix this, k8s setup needs to have power failure protection here. @liyimeng

**Comment by divramod** [comment]
hey @harshavardhana,

i am encountering the same problem with a [kind-cluster](https://kind.sigs.k8s.io/) after restarting my pc.
other deployments are running after the restart, the minio tenants arent't.
you are sure it's about a broken cluster and that it's not a minio problem?

i am getting the same error message as op:
```
E0502 07:23:46.963334       1 cacher.go:419] cacher (*unstructured.Unstructured): unexpected ListAndWatch error: failed to list minio.min.io/v1, Kind=Tenant: conversion webhook for minio.min.io/v2, Kind=Tenant failed: Post "https://operator.minio.svc:4222/webhook/v1/crd-conversion?timeout=30s": dial tcp 10.96.182.177:4222: connect: connection refused; reinitializing...
```

**Comment by dvaldivia** [comment]
that is a kind known problem @divramod I reported that at https://github.com/kubernetes-sigs/kind/issues/1990 a while back, they tried to fix it in a v0.11.0 but I still experience the problem with kind upon restart of the machine, my solution is to tear down my kind cluster and rebuild it.

Perhaps you can open a new issue a report the problem that the kind network is still breaking after machine restart/going to sleep.
